sql/parser/builtins.go

```diff
@@ -943,6 +961,125 @@ func datumToString(datum Datum) (string, error) {
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
 
+var replaceSubRe = regexp.MustCompile(`\\[&1-9]`)
+
+func regexpReplace(s, pattern, to, sqlFlags string) (Datum, error) {
+	pattern, global, err := regexpEvalFlags(pattern, sqlFlags)
+	if err != nil {
+		return nil, err
+	}
+
+	patternRe, err := regexp.Compile(pattern)
```

Comment:
That makes a lot of sense. A follow up question would then be if it's even possible to cache these regexps for an entire query? Given a query like: `SELECT regexp_replace('1', text(generate_series),'9') FROM (SELECT * FROM generate_series(0, 3)) AS t;` wouldn't the pattern be different for each row?